src/pytorch_lightning/tuner/tuning.py

```diff
@@ -212,3 +262,34 @@ def lr_find(
         )
         self.trainer.auto_lr_find = False
         return result["lr_find"]
+
+
+def _check_tuner_configuration(
+    trainer: "pl.Trainer",
+    train_dataloaders: Optional[Union[TRAIN_DATALOADERS, "pl.LightningDataModule"]] = None,
+    val_dataloaders: Optional[EVAL_DATALOADERS] = None,
+    dataloaders: Optional[EVAL_DATALOADERS] = None,
+    method: str = "fit",
+) -> None:
+    supported_methods = ("fit", "validate", "test", "predict")
+    if method not in supported_methods:
+        raise MisconfigurationException(f"method {method!r} is invalid. Should be one of {supported_methods}.")
+
+    if method == "fit":
+        if dataloaders is not None:
+            raise MisconfigurationException(
+                f"In tuner with method={method!r}, `dataloaders` argument should be None,"
+                " please consider setting `train_dataloaders` and `val_dataloaders` instead."
+            )
+    else:
+        if train_dataloaders is not None or val_dataloaders is not None:
+            raise MisconfigurationException(
+                f"In tuner with `method`={method!r}, `train_dataloaders` and `val_dataloaders`"
+                " arguments should be None, please consider setting `dataloaders` instead."
+            )
```

Comment:
Wouldn't these validations be covered by the code added in https://github.com/Lightning-AI/lightning/pull/14637?

Reply:
this is a little different. here we are accepting both `train/val_dataloaders` and `dataloaders`. So if a user passes all of them for some reason, this error is being raised.